app/controllers/users_controller.rb

```diff
@@ -20,19 +22,33 @@ def unlock_access
     redirect_to sufia.profile_path(@user.to_param)
   end
  
-  def link_account
-  end
+  def link_account; end
 
   def set_saml
-    if (params[:has_account] == 'no') || params[:user][:email].nil?
-      @user.ccid = @user.email
-      flash[:notice] = I18n.t('devise.omniauth_callbacks.success', :kind => 'Shibboleth')
-      sign_in_and_redirect @user, :event => :authentication
+    if params[:has_account] == 'yes'
+      existing_user = User.find_by_email(params[:user][:email]) if params[:user][:email].present?
+
+      # the user may have typoed the email they tried to link
+      # send them back to try again
```

Comment:
Makes sense. In order to make it act the same in the 'found actual user' and 'you literally typed gibberish' states, we will need to delete the CCID user regardless, and recreate it the next time they try to log in from shibboleth.